app/src/main/java/net/cassiolandim/florianopolisroutes/activity/ListActivity.java

```diff
@@ -0,0 +1,204 @@
+public class ListActivity extends ParentActivity {
+
+    public static final int PICK_STREET_NAME = 0;
+
+    private EditText mStreetNameEditText;
+    private ListView mListView;
+    private ProgressBar mLoadingSpinner;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_list);
+        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+
+        mLoadingSpinner = (ProgressBar) findViewById(R.id.loading_spinner);
+        mListView = (ListView) findViewById(R.id.list_view);
+        mStreetNameEditText = (EditText) findViewById(R.id.street_name);
+
+        mLoadingSpinner.setVisibility(View.GONE);
+
+        final AdapterView.OnItemClickListener mMessageClickedHandler = new AdapterView.OnItemClickListener() {
+            public void onItemClick(AdapterView parent, View v, int position, long id) {
+                RouteListItem item = (RouteListItem) parent.getAdapter().getItem(position);
+                Intent intent = new Intent(ListActivity.this, DetailsActivity.class);
+                intent.putExtra(DetailsActivity.EXTRA_ID, item.id);
+                intent.putExtra(DetailsActivity.EXTRAS_NAME, item.shortName + " - " + item.longName);
+                startActivity(intent);
+            }
+        };
+
+        mListView.setOnItemClickListener(mMessageClickedHandler);
+
+        findViewById(R.id.search_button).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                doSearch(mStreetNameEditText.getText().toString());
+            }
+        });
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        getMenuInflater().inflate(R.menu.menu_main, menu);
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        int id = item.getItemId();
+
+        if (id == R.id.action_map) {
+            Intent intent = new Intent(ListActivity.this, MapsActivity.class);
+            startActivityForResult(intent, PICK_STREET_NAME);
+            return true;
+        }
+
+        return super.onOptionsItemSelected(item);
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        if (requestCode == PICK_STREET_NAME) {
+            if (resultCode == RESULT_OK) {
+                String streetName = data.getStringExtra(MapsActivity.EXTRA_STREET_NAME);
+                mStreetNameEditText.setText(streetName);
+                doSearch(streetName);
+            }
+        }
+    }
+
+    private void doSearch(final String streetName) {
+        Call<RestApiFindRoutesResponse> call = buildFindRoutesByStopNameCall(streetName);
+
+        mLoadingSpinner.setVisibility(View.VISIBLE);
+        mListView.setVisibility(View.GONE);
+
+        call.enqueue(new Callback<RestApiFindRoutesResponse>() {
```

Comment:
In a more complex app it would be nice to have a better separation of concerns through the Model View Presenter pattern.